azure/services/managedclusters/spec.go

```diff
@@ -487,7 +487,14 @@ func (s *ManagedClusterSpec) Parameters(ctx context.Context, existing *asocontai
 				Key:  secret.KubeconfigDataName,
 			},
 		},
+		ConfigMaps: &asocontainerservicev1.ManagedClusterOperatorConfigMaps{
```

Comment:
I think we want to move this entire block down into the block at L591

Something like 

```go
        if s.OIDCIssuerProfile != nil {
		managedCluster.Spec.OidcIssuerProfile = &asocontainerservicev1.ManagedClusterOIDCIssuerProfile{
		    Enabled: s.OIDCIssuerProfile.Enabled,
		}
                if s.OIDCIssuerProfile.Enabled {
                    managedCluster.Spec.OperatorSpec.ConfigMaps = &asocontainerservicev1.ManagedClusterOperatorConfigMaps{
                        OIDCIssuerProfile: &genruntime.ConfigMapDestination{
			    Name: oidcIssuerURLConfigMapName(s.ClusterName),
			    Key:  oidcIssuerProfileURL,
			},
                    }
                }
	}
```

This way we don't ship a OIDCIssuer URL config when oidc is disabled.

Reply:
Thanks for pointing it out. I agree that we should move this block down.

Reply:
Webhooks check out, OIDC Issuer cannot be disable once enabled. 
Defaulting webhook: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/5e1e3283b38c068797e46fd361b52483717eebf1/api/v1beta1/azuremanagedcontrolplane_default.go#L195-L203

Validating webhook: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/5e1e3283b38c068797e46fd361b52483717eebf1/api/v1beta1/azuremanagedcontrolplane_webhook.go#L812-L826

Reply:
added via : 43d109d495aa3cf36a286b00fc1db95e6157e40a